lib/events/dynamoevents/dynamoevents.go

```diff
@@ -1235,6 +1235,39 @@ func (l *Log) convertFieldsToDynamoMapFormat(ctx context.Context) error {
 	return trace.Wrap(l.migrateMatchingEvents(ctx, filterExpr, transformEvent))
 }
 
+func minInt64(x, y int64) int64 {
+	if x > y {
+		return y
+	}
+	return x
+}
+
+func maxInt64(x, y int64) int64 {
+	if x > y {
+		return x
+	}
+	return y
+}
+
+func (l *Log) approximateOptimalMigrationWorkers() (int64, error) {
+	req := dynamodb.DescribeTableInput{TableName: aws.String(l.Tablename)}
+	table, err := l.svc.DescribeTable(&req)
+	if err != nil {
+		return 0, trace.Wrap(err)
+	}
+
+	// calculate the throughput, accounting for r/w bottlenecks
+	provisioned := table.Table.ProvisionedThroughput
+	if provisioned == nil || provisioned.ReadCapacityUnits == nil || provisioned.WriteCapacityUnits == nil {
+		return maxMigrationWorkers, nil
+	}
+	throughput := minInt64(*provisioned.ReadCapacityUnits, *provisioned.WriteCapacityUnits)
+
+	// divide throughput by batch size rounding upwards and then take 75% of that
+	optimalWorkers := (throughput + (DynamoBatchSize - 1)) / DynamoBatchSize * 3 / 4
```

Comment:
```suggestion
	optimalWorkers := (throughput + (DynamoBatchSize - 1)) / DynamoBatchSize * 0.75
```
to match your comment?